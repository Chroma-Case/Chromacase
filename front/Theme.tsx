import { NativeBaseProvider, extendTheme, useColorMode } from 'native-base';
import useColorScheme from './hooks/colorScheme';
import { useEffect } from 'react';
import { Platform } from 'react-native';

const defaultLightGlassmorphism = {
	50: 'rgba(255,255,255,0.9)',
	100: 'rgba(255,255,255,0.1)',
	200: 'rgba(255,255,255,0.2)',
	300: 'rgba(255,255,255,0.3)',
	400: 'rgba(255,255,255,0.4)',
	500: 'rgba(255,255,255,0.5)',
	600: 'rgba(255,255,255,0.6)',
	700: 'rgba(255,255,255,0.7)',
	800: 'rgba(255,255,255,0.8)',
	900: 'rgba(255,255,255,0.9)',
	1000: 'rgba(255,255,255,1)',
};
// shodws are hugely visible on phone so we trick the colors without alpha
const phoneLightGlassmorphism = {
	50: 'rgb(200, 204, 254)',
	100: 'rgb(204, 208, 254)',
	200: 'rgb(210, 214, 254)',
	300: 'rgb(214, 218, 254)',
	400: 'rgb(220, 222, 254)',
	500: 'rgb(230, 234, 254)',
	600: 'rgb(234, 236, 254)',
	700: 'rgb(240, 242, 254)',
	800: 'rgb(244, 246, 254)',
	900: 'rgb(248, 250, 254)',
	1000: 'rgb(252, 254, 254)',
};
const lightGlassmorphism =
	Platform.OS === 'web' ? defaultLightGlassmorphism : phoneLightGlassmorphism;
const darkGlassmorphism = {
	50: 'rgba(16,16,20,0.9)',
	100: 'rgba(16,16,20,0.1)',
	200: 'rgba(16,16,20,0.2)',
	300: 'rgba(16,16,20,0.3)',
	400: 'rgba(16,16,20,0.4)',
	500: 'rgba(16,16,20,0.5)',
	600: 'rgba(16,16,20,0.6)',
	700: 'rgba(16,16,20,0.7)',
	800: 'rgba(16,16,20,0.8)',
	900: 'rgba(16,16,20,0.9)',
	1000: 'rgba(16,16,20,1)',
};

const ThemeProvider = ({ children }: { children: JSX.Element }) => {
	const colorScheme = useColorScheme();

	const glassmorphism = colorScheme === 'light' ? lightGlassmorphism : darkGlassmorphism;
	const text = colorScheme === 'light' ? darkGlassmorphism : lightGlassmorphism;

	return (
		<NativeBaseProvider
			isSSR={false}
			theme={extendTheme({
				config: {
					useSystemColorMode: false,
					initialColorMode: colorScheme,
				},
				fonts: {
					heading: 'Lexend',
					body: 'Lexend',
					mono: 'Lexend',
				},
				colors: {
					text: text,
					primary: {
						50: '#eff1fe',
						100: '#e7eafe',
						200: '#cdd4fd',
						300: '#5f74f7',
						400: '#5668de',
						500: '#4c5dc6',
						600: '#4757b9',
						700: '#394694',
						800: '#2b346f',
						900: '#212956',
					},
					secondary: {
						50: '#f7f3ff',
						100: '#f3edfe',
						200: '#e6d9fe',
						300: '#ae84fb',
						400: '#9d77e2',
						500: '#8b6ac9',
						600: '#8363bc',
						700: '#684f97',
						800: '#4e3b71',
						900: '#3d2e58',
					},
					error: {
						50: '#f7f3ff',
						100: '#f3edfe',
						200: '#e6d9fe',
						300: '#ae84fb',
						400: '#9d77e2',
						500: '#8b6ac9',
						600: '#8363bc',
						700: '#684f97',
						800: '#4e3b71',
						900: '#3d2e58',
					},
					alert: {
						50: '#fff2f1',
						100: '#ffebea',
						200: '#ffd6d3',
						300: '#ff7a72',
						400: '#e66e67',
						500: '#cc625b',
						600: '#bf5c56',
						700: '#994944',
						800: '#733733',
						900: '#592b28',
					},
					notification: {
						50: '#fdfbec',
						100: '#fcf9e2',
						200: '#f8f3c3',
						300: '#ead93c',
						400: '#d3c336',
						500: '#bbae30',
						600: '#b0a32d',
						700: '#8c8224',
						800: '#69621b',
						900: '#524c15',
					},
					black: {
						50: '#e7e7e8',
						100: '#dbdbdc',
						200: '#b5b5b6',
						300: '#101014',
						400: '#0e0e12',
						500: '#0d0d10',
						600: '#0c0c0f',
						700: '#0a0a0c',
						800: '#070709',
						900: '#060607',
					},
					red: {
						50: '#fdedee',
						100: '#fce4e5',
						200: '#f9c7c9',
						300: '#ed4a51',
						400: '#d54349',
						500: '#be3b41',
						600: '#b2383d',
						700: '#8e2c31',
						800: '#6b2124',
						900: '#531a1c',
					},
					coolGray: glassmorphism,
				},
				components: {
					Button: {
						baseStyle: () => ({
							borderRadius: 'md',
						}),
					},
					Link: {
						defaultProps: {
							isUnderlined: false,
						},
						baseStyle: () => ({
							_text: {
								color: 'secondary.300',
							},
							_hover: {
								isUnderlined: true,
								_text: {
									color: 'secondary.400',
								},
							},
						}),
					},
				},
			})}
		>
			{children}
		</NativeBaseProvider>
	);
};

const ColorSchemeProvider = (props: { children: JSX.Element }) => {
	const colorScheme = useColorScheme();
	const colorMode = useColorMode();

	useEffect(() => {
		colorMode.setColorMode(colorScheme);
	}, [colorScheme]);
	return props.children;
};

export default ThemeProvider;
export { ColorSchemeProvider };
