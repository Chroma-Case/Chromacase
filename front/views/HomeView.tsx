import React from 'react';
import { useQueries, useQuery } from '../Queries';
import API from '../API';
import { LoadingView } from '../components/Loading';
import { Box, Flex, Stack, Heading, VStack, HStack } from 'native-base';
import { RouteProps, useNavigation } from '../Navigation';
import SongCardGrid from '../components/SongCardGrid';
import CompetenciesTable from '../components/CompetenciesTable';
import Translate from '../components/Translate';
import TextButton from '../components/TextButton';
import Song from '../models/Song';
import { FontAwesome5 } from '@expo/vector-icons';
import ScaffoldCC from '../components/UI/ScaffoldCC';

// eslint-disable-next-line @typescript-eslint/ban-types
const HomeView = (props: RouteProps<{}>) => {
	const navigation = useNavigation();
	const userQuery = useQuery(API.getUserInfo);
	const playHistoryQuery = useQuery(API.getUserPlayHistory);
	const searchHistoryQuery = useQuery(API.getSearchHistory(0, 10));
	const skillsQuery = useQuery(API.getUserSkills);
	const nextStepQuery = useQuery(API.getSongSuggestions);
	const songHistory = useQueries(
		playHistoryQuery.data?.map(({ songID }) => API.getSong(songID)) ?? []
	);
	const artistsQueries = useQueries(
		songHistory
			.map((entry) => entry.data)
			.concat(nextStepQuery.data ?? [])
			.filter((s): s is Song => s !== undefined)
			.map((song) => API.getArtist(song.artistId))
	);

	if (
		!userQuery.data ||
		!skillsQuery.data ||
		!searchHistoryQuery.data ||
		!playHistoryQuery.data
	) {
		return <LoadingView />;
	}
	return (
		<ScaffoldCC routeName={props.route.name}>
			<Stack direction={{ base: 'column', lg: 'row' }} space={5} paddingTop={5}>
				<VStack flex={{ lg: 2 }} space={5}>
					<SongCardGrid
						heading={<Translate translationKey="goNextStep" />}
						songs={
							nextStepQuery.data
								?.filter((song) =>
									artistsQueries.find(
										(artistQuery) => artistQuery.data?.id === song.artistId
									)
								)
								.map((song) => ({
									cover: song.cover,
									name: song.name,
									songId: song.id,
									artistName: artistsQueries.find(
										(artistQuery) => artistQuery.data?.id === song.artistId
									)!.data!.name,
								})) ?? []
						}
					/>
					<Stack direction={{ base: 'column', lg: 'row' }}>
						<Box flex={{ lg: 1 }}>
							<Heading>
								<Translate translationKey="mySkillsToImprove" />
							</Heading>
							<Box padding={5}>
								<CompetenciesTable {...skillsQuery.data} />
							</Box>
						</Box>
						<Box flex={{ lg: 1 }}>
							<SongCardGrid
								heading={<Translate translationKey="recentlyPlayed" />}
								songs={
									songHistory
										.map(({ data }) => data)
										.filter((data): data is Song => data !== undefined)
										.filter(
											(song, i, array) =>
												array
													.map((s) => s.id)
													.findIndex((id) => id == song.id) == i
										)
										.filter((song) =>
											artistsQueries.find(
												(artistQuery) =>
													artistQuery.data?.id === song.artistId
											)
										)
										.map((song) => ({
											cover: song.cover,
											name: song.name,
											songId: song.id,
											artistName: artistsQueries.find(
												(artistQuery) =>
													artistQuery.data?.id === song.artistId
											)!.data!.name,
										})) ?? []
								}
							/>
						</Box>
					</Stack>
				</VStack>
				<VStack flex={{ lg: 1 }} height={{ lg: '100%' }} alignItems="center">
					<HStack width="100%" justifyContent="space-evenly" p={5} space={5}>
						<TextButton
							translate={{ translationKey: 'searchBtn' }}
							colorScheme="secondary"
							size="sm"
							onPress={() => navigation.navigate('Search', {})}
						/>
						<TextButton
							translate={{ translationKey: 'settingsBtn' }}
							colorScheme="gray"
							size="sm"
							onPress={() => navigation.navigate('Settings', {})}
						/>
						<TextButton
							label={'V2'}
							colorScheme="gray"
							size="sm"
							onPress={() => navigation.navigate('HomeNew', {})}
						/>
					</HStack>
					<Box style={{ width: '100%' }}>
						<Heading>
							<Translate translationKey="recentSearches" />
						</Heading>
						<Flex
							padding={3}
							style={{
								width: '100%',
								alignItems: 'flex-start',
								alignContent: 'flex-start',
								flexDirection: 'row',
								flexWrap: 'wrap',
							}}
						>
							{searchHistoryQuery.data?.length === 0 && (
								<Translate translationKey="noRecentSearches" />
							)}
							{[...new Set(searchHistoryQuery.data.map((x) => x.query))]
								.slice(0, 5)
								.map((query) => (
									<TextButton
										leftIcon={<FontAwesome5 name="search" size={16} />}
										style={{
											margin: 2,
										}}
										key={query}
										variant="solid"
										size="xs"
										colorScheme="primary"
										label={query}
										onPress={() =>
											navigation.navigate('Search', { query: query })
										}
									/>
								))}
						</Flex>
					</Box>
				</VStack>
			</Stack>
		</ScaffoldCC>
	);
};

export default HomeView;
