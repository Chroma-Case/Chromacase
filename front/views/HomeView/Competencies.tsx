import React from "react"
import { Card, Text } from "react-native-paper"
import { View } from "react-native"
import ProgressBar from "../../components/progressBar"
import { AvailableLanguages, translate } from "../../i18n/i18n";
import { useSelector } from "react-redux";

const cardFormat = {
    marginLeft: 20,
    width: 700,
    height: 400,
}

const textStyle = {
  fontSize: 20,
  padding: 20,
}

const rowStyle = {
  flexDirection: 'row',
  justifyContent: 'space-between'
}

const Competencies = () => {
  const language: AvailableLanguages = useSelector((state) => state.language.value);

  return (
    <Card style={cardFormat}>
      <Card.Content style={cardFormat}>
        <View style={rowStyle}>
          <Text style={textStyle}> { translate('pedalsCompetency') } </Text>
          <ProgressBar progress={60} maxValue={100} barWidth={70}/>
        </View>
        <View style={rowStyle}>
          <Text style={textStyle}> { translate('rightHandCompetency') } </Text>
          <ProgressBar progress={80} maxValue={100} barWidth={70}/>
        </View>
        <View style={rowStyle}>
          <Text style={textStyle}> { translate('leftHandCompetency') } </Text>
          <ProgressBar progress={30} maxValue={100} barWidth={70}/>
        </View>
        <View style={rowStyle}>
          <Text style={textStyle}> { translate('accuracyCompetency') } </Text>
          <ProgressBar progress={30} maxValue={100} barWidth={70}/>
        </View>
        <View style={rowStyle}>
          <Text style={textStyle}> { translate('arpegeCompetency') } </Text>
          <ProgressBar progress={70} maxValue={100} barWidth={70}/>
        </View>
        <View style={rowStyle}>
          <Text style={textStyle}> { translate('chordsCompetency') } </Text>
          <ProgressBar progress={80} maxValue={100} barWidth={70}/>
        </View>
      </Card.Content>
    </Card>
  );
}

export default Competencies;