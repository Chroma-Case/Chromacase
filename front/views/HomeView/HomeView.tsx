import React from "react";
import { useQuery } from "react-query";
import API from "../../API";
import LoadingComponent from "../../components/Loading";
import { Box, ScrollView, useBreakpointValue, Text, VStack, Progress, Button, HStack } from 'native-base';
import SongCard from "../../components/SongCard";
import { FlatGrid } from 'react-native-super-grid';

const ProgressBar = ({ xp }: { xp: number}) => {
	const level = Math.floor(xp / 1000);
	const nextLevel = level + 1;
	const nextLevelThreshold = nextLevel * 1000;
	const progessValue = 100 * xp / nextLevelThreshold;
	return <VStack alignItems={'center'}>
		<Text>Niveau {level}</Text>
		<Box w="90%" maxW="400">
			<Progress value={progessValue} mx="4" />
		</Box>
		<Text>{xp} / {nextLevelThreshold} bonnes notes</Text>
	</VStack>
}

const HomeView = () => {
	const flexDirection = useBreakpointValue({ base: 'column', xl: "row"});
	const userQuery = useQuery(['user'], () => API.getUserInfo());
	if (!userQuery.data) {
		return <Box style={{ flexGrow: 1, justifyContent: 'center' }}>
			<LoadingComponent/>
		</Box>
	}
	return <ScrollView>
		<Box style={{ display: 'flex', padding: 10 }}>
			<Box textAlign={ flexDirection == 'column' ? 'center' : undefined } style={{ flexDirection, justifyContent: 'center', display: 'flex' }}>
				<Text fontSize="xl" flex={1}>Bienvenue {userQuery.data.name}!</Text>
				<Box flex={1}>
					<ProgressBar xp={userQuery.data.xp}/>
				</Box>
			</Box>
			<Box style={{ flexDirection }}>
				<Box flex={1}>
					<Text fontSize="md">Passer à l'étape supérieure</Text>
					<FlatGrid	
						data={[ ...Array(4).keys() ]}
						renderItem={({ item }) =>
							<SongCard albumCover={"https://meelo.arthichaud.me/api/illustrations/releases/120"} songTitle={"Song"} artistName={"Artist"}/>
						}
						spacing={20}
					/>
				</Box>
				<VStack flex={1} >
					<Box style={{ flexDirection: 'row', justifyContent:'center' }}>
						<Button size="sm">Search</Button>

					</Box>
				</VStack>
			</Box>
		</Box>
	</ScrollView>
	
}

export default HomeView;
