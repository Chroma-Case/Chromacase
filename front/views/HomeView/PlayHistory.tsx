import React from "react";
import Col from "../../components/col";
import Row from "../../components/row";
import SongCard from "../../components/songCard";

const RecentlyPlayed = () => {
    return (
      <Col>
        <Row>
          <SongCard albumCover="https://e-cdn-images.dzcdn.net/images/artist/61bcbf8296b1669499064406c534d39d/264x264-000000-80-0-0.jpg"
                    songTitle="Stay foolish"
                    artistName="ZUTOMAYO"/>
          <SongCard albumCover="https://e-cdn-images.dzcdn.net/images/artist/61bcbf8296b1669499064406c534d39d/264x264-000000-80-0-0.jpg"
                    songTitle="Mirror tune"
                    artistName="ZUTOMAYO"/>
        </Row>
        <Row>
          <SongCard albumCover="https://e-cdn-images.dzcdn.net/images/artist/61bcbf8296b1669499064406c534d39d/264x264-000000-80-0-0.jpg"
                    songTitle="Neko reset"
                    artistName="ZUTOMAYO"/>
          <SongCard albumCover="https://e-cdn-images.dzcdn.net/images/artist/61bcbf8296b1669499064406c534d39d/264x264-000000-80-0-0.jpg"
                    songTitle="Milabo"
                    artistName="ZUTOMAYO"/>
        </Row>
      </Col>
    );
}

export default RecentlyPlayed;