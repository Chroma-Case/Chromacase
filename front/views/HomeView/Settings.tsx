import React from "react";
import { Button, Text } from "react-native-paper";
import { useDispatch, useSelector } from "react-redux";
import { AvailableLanguages, translate, DefaultLanguage } from "../../i18n/i18n";
import { useLanguage } from "../../state/LanguageSlice";
import { unsetUserToken } from "../../state/UserSlice";
import { useNavigation } from "@react-navigation/native";

const SettingsButton = () => {
    const dispatch = useDispatch();
    const navigation = useNavigation();
    const language: AvailableLanguages = useSelector((state) => state.language.value);
  
    return (
      <div>
        <Button onPress={() => dispatch(unsetUserToken())}>
          { translate('signoutBtn') }
        </Button>
        <Button onPress={() => {
          let newLanguage = DefaultLanguage;
          switch (language) {
            case 'en':
              newLanguage = 'fr';
              break;
            default:
              break;
          }
          dispatch(useLanguage(newLanguage));
        }}>
          { translate('changeLanguageBtn') }
        </Button>
        <Text style={{ textAlign: "center" }}>Current language: { language }</Text>
        <Button onPress={() =>  navigation.navigate('Song', { songId: 1 }) }> { translate('songPageBtn') }</Button>
      </div>
    )
}

export default SettingsButton;