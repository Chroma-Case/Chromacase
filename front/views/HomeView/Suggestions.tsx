import React from "react";
import SongCard from "../../components/songCard";
import Row from "../../components/row";

const Suggestions = () => {
    return (
      <Row>
        <SongCard albumCover="https://i.discogs.com/yHqu3pnLgJq-cVpYNVYu6mE-fbzIrmIRxc6vES5Oi48/rs:fit/g:sm/q:90/h:556/w:600/czM6Ly9kaXNjb2dz/LWRhdGFiYXNlLWlt/YWdlcy9SLTE2NjQ2/ODUwLTE2MDkwNDU5/NzQtNTkxOS5qcGVn.jpeg"
                  songTitle="Dramaturgy"
                  artistName="Eve MV"/>
        <SongCard albumCover="https://i.discogs.com/yHqu3pnLgJq-cVpYNVYu6mE-fbzIrmIRxc6vES5Oi48/rs:fit/g:sm/q:90/h:556/w:600/czM6Ly9kaXNjb2dz/LWRhdGFiYXNlLWlt/YWdlcy9SLTE2NjQ2/ODUwLTE2MDkwNDU5/NzQtNTkxOS5qcGVn.jpeg"
                  songTitle="Shinkai"
                  artistName="Eve MV"/>
        <SongCard albumCover="https://i.discogs.com/yHqu3pnLgJq-cVpYNVYu6mE-fbzIrmIRxc6vES5Oi48/rs:fit/g:sm/q:90/h:556/w:600/czM6Ly9kaXNjb2dz/LWRhdGFiYXNlLWlt/YWdlcy9SLTE2NjQ2/ODUwLTE2MDkwNDU5/NzQtNTkxOS5qcGVn.jpeg"
                  songTitle="Don't replay the boredom"
                  artistName="Eve MV"/>
        <SongCard albumCover="https://i.discogs.com/yHqu3pnLgJq-cVpYNVYu6mE-fbzIrmIRxc6vES5Oi48/rs:fit/g:sm/q:90/h:556/w:600/czM6Ly9kaXNjb2dz/LWRhdGFiYXNlLWlt/YWdlcy9SLTE2NjQ2/ODUwLTE2MDkwNDU5/NzQtNTkxOS5qcGVn.jpeg"
                  songTitle="Heart forecast"
                  artistName="Eve MV"/>
      </Row>
    );
}

export default Suggestions;