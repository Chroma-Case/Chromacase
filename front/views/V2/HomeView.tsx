import { View } from 'react-native';
import { Text, useBreakpointValue } from 'native-base';
import React from 'react';
import { useQuery } from '../../Queries';
import HomeMainSongCard from '../../components/V2/HomeMainSongCard';
import SongCardInfo from '../../components/V2/SongCardInfo';
import API from '../../API';

const bigSideRatio = 1000;
const smallSideRatio = 618;

type HomeCardProps = {
	image: string;
	title: string;
	artist: string;
};

const cards = [
	{
		image: 'https://media.discordapp.net/attachments/717080637038788731/1153688155292180560/image_homeview1.png',
		title: 'Beethoven',
		artist: 'Synphony No. 9',
	},
	{
		image: 'https://media.discordapp.net/attachments/717080637038788731/1153688154923090093/image_homeview2.png',
		title: 'Mozart',
		artist: 'Lieder Kantate KV 619',
	},
	{
		image: 'https://media.discordapp.net/attachments/717080637038788731/1153688154499457096/image_homeview3.png',
		title: 'Back',
		artist: 'Truc Truc',
	},
	{
		image: 'https://media.discordapp.net/attachments/717080637038788731/1153688154109394985/image_homeview4.png',
		title: 'Mozart',
		artist: 'Machin Machin',
	},
] as [HomeCardProps, HomeCardProps, HomeCardProps, HomeCardProps];

const HomeView = () => {
	const songsQuery = useQuery(API.getSongSuggestions);
	const screenSize = useBreakpointValue({ base: 'small', md: 'big' });
	const isPhone = screenSize === 'small';

	return (
		<View
			style={{
				width: '100%',
				height: '100%',
				display: 'flex',
				flexDirection: 'column',
			}}
		>
			<View>
				<View
					style={{
						alignSelf: 'stretch',
						maxWidth: '1100px',
						alignItems: 'stretch',
						flexDirection: isPhone ? 'column' : 'row',
					}}
				>
					<View
						style={{
							flexGrow: bigSideRatio,
						}}
					>
						<HomeMainSongCard {...cards[0]} />
					</View>
					<View
						style={{
							flexGrow: smallSideRatio,
							display: 'flex',
							flexDirection: isPhone ? 'row' : 'column',
							alignItems: 'stretch',
						}}
					>
						<View
							style={{
								flexGrow: bigSideRatio,
							}}
						>
							<HomeMainSongCard {...cards[1]} />
						</View>
						<View
							style={{
								flexGrow: smallSideRatio,
								display: 'flex',
								flexDirection: isPhone ? 'column-reverse' : 'row-reverse',
								alignItems: 'stretch',
							}}
						>
							<View
								style={{
									flexGrow: bigSideRatio,
								}}
							>
								<HomeMainSongCard {...cards[2]} />
							</View>
							<View
								style={{
									flexGrow: smallSideRatio,
									display: 'flex',
									flexDirection: isPhone ? 'row-reverse' : 'column-reverse',
									alignItems: 'stretch',
								}}
							>
								<View
									style={{
										flexGrow: bigSideRatio,
										display: 'flex',
										flexDirection: 'column',
										alignItems: 'stretch',
										justifyContent: 'flex-end',
									}}
								>
									<HomeMainSongCard {...cards[3]} />
								</View>
								<View
									style={{
										flexGrow: smallSideRatio,
									}}
								></View>
							</View>
						</View>
					</View>
				</View>
			</View>
			<View
				style={{
					flexShrink: 0,
					flexGrow: 0,
					flexBasis: '15%',
					width: '100%',
				}}
			>
				<Text
					style={{
						color: 'white',
						fontSize: 24,
						fontWeight: 'bold',
						marginLeft: 16,
						marginBottom: 16,
						marginTop: 24,
					}}
				>
					{'Suggestions'}
				</Text>
				{songsQuery.isLoading && <Text>Loading...</Text>}
				<View
					style={{
						flexDirection: 'row',
						flexWrap: 'wrap',
						justifyContent: 'flex-start',
						alignItems: 'flex-start',
						// @ts-expect-error - gap is not in the typings
						gap: 16,
					}}
				>
					{songsQuery.data?.map((song) => (
						<SongCardInfo
							key={song.id}
							song={song}
							onPress={() => {
								console.log('SongCardInfo pressed');
							}}
						/>
					))}
				</View>
			</View>
		</View>
	);
};

export default HomeView;
