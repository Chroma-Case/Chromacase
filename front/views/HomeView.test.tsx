import React from 'react';
// import { Provider } from 'react-redux';
// import store from '../state/Store';
// import { fireEvent, render, screen } from '@testing-library/react-native';
import { NativeBaseProvider } from "native-base";
import Theme from '../Theme';

import HomeView from './HomeView';
// import { en, fr } from '../i18n/Translations';
import renderer from 'react-test-renderer'


describe('<HomeView />', () => {

  jest.setTimeout(150000);
  // const view = () => <NativeBaseProvider theme={Theme}><HomeView/></NativeBaseProvider>;
  // beforeEach(() => render(view()));

  // it('should render correctly', () => {
  //   expect(screen).toBeDefined();
  // })
  //const View = () => <NativeBaseProvider theme={Theme}><HomeView/></NativeBaseProvider>;

  it('Main View should render correctly', async () => {
    const tree = renderer.create(<NativeBaseProvider theme={Theme}><HomeView/></NativeBaseProvider>).toJSON;
    expect(tree).toMatchSnapshot();
  });


  // it('has should display the text in default language', async () => {
  //   expect((await screen.findAllByText(en.signoutBtn)).length).toBe(1);
  // });

  // it('has should display the text in the new language', async () => {
  //   fireEvent.press(screen.getByText('Change language'));
  //   expect(store.getState().language.value).toBe('fr');
  //   screen.update(view);
  //   expect((await screen.findAllByText(fr.signoutBtn)).length).toBe(1);
  // });
});
