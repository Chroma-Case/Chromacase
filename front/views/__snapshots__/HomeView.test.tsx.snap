// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeView /> Main View should render correctly 1`] = `[Function]`;
